### ISSUE TITLE
kfctl uses app dir in app.yaml and not actual directory 

### ISSUE THREAD

**Issue by jlewi** [kubeflow/kubeflow#3264]
It looks like kfctl is using the application directory in app.yaml and not the directory from which kfctl is running.

This will cause problems when moving the app and when using it with source control.

Example suppose user creates the app in
somedir/myapp

and now they move it to
otherdir/myapp

Now suppose they modify the app e.g. add an NFS deployment manager config to gcp_config
To deploy user runs

```
cd otherdir/myapp
kfctl apply platform
```
Even though kfctl is executed from `otherdir/myapp` it looks like its actually try to read the app from the directory specified in app.yaml

We should fix this. kfctl should respect the working directory.

In the interim, Users can work around this by updating app.yaml.

/cc @gabrielwen @kkasravi @ashahba

**Comment by jlewi** [comment]
It looks like the problem is specific to the handling of extra GCP deployment manager config files
https://github.com/kubeflow/kubeflow/blob/453d96759b7b5615b82a4182555cf82149597a11/bootstrap/pkg/kfapp/gcp/gcp.go#L579

**Comment by jlewi** [comment]
I take that back it looks like we do it for all configs
https://github.com/kubeflow/kubeflow/blob/47a28b7c42689444fdbf12901cbc98125d1908f9/bootstrap/pkg/kfapp/gcp/gcp.go#L311

**Comment by lluunn** [comment]
Should we just remove appDir in spec, and always use pwd?

**Comment by jlewi** [comment]
I think the pattern we want to move to is using KfDef.Status to pass around temporary information in kfctl.
There should be a TODO in the code to move to AppDir into KfDef.Status.

I think it would make sense to set KfDef.Status.AppDir to CWD at suitable entry points in the code.

We should probably wait on #3515 to be submitted first.

**Comment by jlewi** [comment]
I believe this is no longer relevant given all the changes and improvements to kfctl.
